digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala

```diff
@@ -382,29 +402,69 @@ class SpeedLimitsTierekisteriImporter extends LinearAssetTierekisteriImporterOpe
     }
   }
 
-  private def getSpeedLimitValue(trAsset: TierekisteriAssetData) = {
+  private def getSpeedLimitValue(assetInfo: TrAssetInfo, linkType:  Option[LinkType]): Option[Int] = {
+    def getSpeedLimitValueByLinkType(linkType:  Option[LinkType]): Option[Int] = {
+    linkType.getOrElse("") match {
+      case Motorway =>
+        Some(defaultMotorwaySpeedLimit)
+      case MultipleCarriageway | SingleCarriageway | Freeway =>
```

Comment:
Instead of **"linkType.getOrElse("") match {"** why we don't use only the **"linkType match {"** and then we can used like **"case Some(Motorway) =>"**